src/main/java/ca/gc/cyber/ops/assemblyline/java/client/clients/AssemblylineClient.java

```diff
@@ -256,6 +266,16 @@ public Mono<Submission> submitBinary(BinaryFile<SubmitMetadata> binaryIngest) {
                         multipartInserter, MediaType.MULTIPART_FORM_DATA));
     }
 
+    @Override
+    public Mono<Submission> submitAsyncBinary(AsyncBinaryFile<SubmitMetadata> binaryIngest) {
+
+        return Mono.fromCallable(() -> this.multipartInserterFromAsyncBinaryIngest(binaryIngest))
+                .subscribeOn(Schedulers.boundedElastic())
+                .flatMap(multipartInserter -> post(buildUri(SUBMIT_URL), new ParameterizedTypeReference<>() {
+                        },
+                        multipartInserter, MediaType.MULTIPART_FORM_DATA));
+    }
+
```

Comment:
This method seems unlikely to diverge from `submitBinary()`, so you could refactor both methods to reduce duplication.

e.g.
```java
    @Override
    public Mono<Submission> submitBinary(BinaryFile<SubmitMetadata> binaryIngest) {
        return submitBinary(() -> this.multipartInserterFromBinaryIngest(binaryIngest));
    }
    
    @Override
    public Mono<Submission> submitAsyncBinary(AsyncBinaryFile<SubmitMetadata> binaryIngest) {
        return submitBinary(() -> this.multipartInserterFromAsyncBinaryIngest(binaryIngest));
    }
    
    private Mono<Submission> submitBinary(Supplier<BodyInserters.MultipartInserter> bodyInserter) {

        return Mono.fromCallable(bodyInserter)
                .subscribeOn(Schedulers.boundedElastic())
                .flatMap(multipartInserter -> post(buildUri(SUBMIT_URL), new ParameterizedTypeReference<>() {
                        },
                        multipartInserter, MediaType.MULTIPART_FORM_DATA));
    }
```